File: C++/038. Count and Say.cpp
class Solution {
public:
    string transform(string s)
    {
        string res = "";
        int n = s.length();
        int count = 1;
        char c = s[0];
        for (int i = 1; i < n; i++)
        {
            if (s[i] == c)
            {
                count++;
            }
            else
            {
                res += to_string(count);
                res += c;
                c = s[i];
                count = 1;
            }
        }
        res += to_string(count);
        res += c;
        return res;
    }
    string countAndSay(int n) {
        if(n==0) return "";
        string res = "1";
        for (int i = 1; i< n; i++)
        {
            res = transform(res);
        }
        return res;
    }
};

File: python/038. Count and Say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        if n == 1: return "1"
        temp = self.countAndSay(n-1)
        cnt = 0
        cur = ord(temp[0]) - ord('0')
        say = ""
        for digit in temp:
            val = ord(digit) - ord('0')
            if val == cur:
                cnt += 1
            else:
                say += str(cnt)
                say += str(cur)
                cur = val
                cnt = 1
        say += str(cnt)
        say += str(cur)
        return say
